mitiq/pec/representations/tests/test_depolarizing.py

```diff
@@ -175,3 +179,73 @@ def test_three_qubit_local_depolarizing_representation_error():
             Circuit(CCNOT(q0, q1, q2)),
             0.05,
         )
+
+
+@pytest.mark.parametrize("circuit_type", ["cirq", "qiskit", "pyquil"])
+def test_represent_operations_in_circuit_global(circuit_type: str):
+    """Tests all operation representations are created."""
+    qreg = LineQubit.range(2)
+    circ = Circuit([CNOT(*qreg), H(qreg[0]), Y(qreg[1]), CNOT(*qreg)])
+    reps = represent_operations_in_circuit_with_global_depolarizing_noise(
+        ideal_circuit=circ,
+        noise_level=0.1,
+    )
+    # For each operation in circ we should find its representation
+    for op in circ.all_operations():
+        found = False
+        for rep in reps:
+            if _equal(rep.ideal, Circuit(op), require_qubit_equality=True):
+                found = True
+        assert found
+
+    # The number of reps. should match the number of unique operations
+    assert len(reps) == 3
+
+
+@pytest.mark.parametrize("circuit_type", ["cirq", "pyquil"])
+def test_represent_operations_in_circuit_local(circuit_type: str):
+    """Tests all operation representations are created."""
+    qreg = LineQubit.range(2)
+    circ_mitiq = Circuit([CNOT(*qreg), H(qreg[0]), Y(qreg[1]), CNOT(*qreg)])
+    circ = convert_from_mitiq(circ_mitiq, circuit_type)
+
+    reps = represent_operations_in_circuit_with_local_depolarizing_noise(
+        ideal_circuit=circ,
+        noise_level=0.1,
+    )
+
+    for op in convert_to_mitiq(circ)[0].all_operations():
+        found = False
+        for rep in reps:
+            if _equal(rep.ideal, Circuit(op), require_qubit_equality=True):
+                found = True
+        assert found
+
+    # The number of reps. should match the number of unique operations
+    assert len(reps) == 3
+
+
+# TODO: Once the problem with qubit names is fixed, remove the next
+# test and add "qiskit" circuit type the test above.
+@pytest.mark.skip(reason="Problem with conversions of qiskit qubit names")
+@pytest.mark.parametrize("circuit_type", ["qiskit"])
```

Comment:
Can you elaborate on this?

Reply:
I made an issue about this (see  #516 )

Reply:
Update after this commit: https://github.com/unitaryfund/mitiq/pull/515/commits/5c60c11e9efeffd9ee41fbdc12b342b60cd155a1

The qiskit test is now added and working fine, at the price of
`represent_operations_in_circuit_with_...`  returning always representations where  `. native_ideal` is a `cirq.Circuit`.

This is not an ideal solution but I think that, because of #516, it is hard to return the correct native operation (with correct qubit names).

Reply:
Hmm, maybe this isn't so bad. All PEC types are supposed to be platform agnostic. 

First, I realized from this that [`OperationRepresentation.ideal`](https://github.com/unitaryfund/mitiq/blob/master/mitiq/pec/types/types.py#L410) should be rewritten like [`NoisyOperation.ideal_circuit`](https://github.com/unitaryfund/mitiq/blob/master/mitiq/pec/types/types.py#L190). Currently `OperationRepresentation._native_ideal` isn't used anywhere.

Second, with the pattern of `represent_operations_in_circuit*` functions, I think we will always be converting the input `QPROGRAM` to a `cirq.Circuit` to iterate over the operations. If this is the case, maybe `OperationRepresentation` should be redesigned a bit to support setting the native type. Something roughly like

```python
def represent_operations_in_circuit*(circuit: QPROGRAM):
    circuit, ctype = convert_to_mitiq(circuit)

    for op in set(circuit.all_operations()):
        rep = OperationRepresentation(Circuit(op), basis_expansion, native_type=ctype)
```
 What do you think? Does this fix all the issues here?

Reply:
Note that at the moment we use the fact that [`OperationRepresentation.ideal`](https://github.com/unitaryfund/mitiq/blob/master/mitiq/pec/types/types.py#L410) always returns Cirq circuits.
This is used [sampling_sequence](https://github.com/unitaryfund/mitiq/blob/74253d72da759d290c8c9cf7273108df79aeb1cb/mitiq/pec/sampling.py#L67). If we change the return type of `OperationRepresentation.ideal` we need to reconsider this sampling step.

I am not sure if adding a `native_type` argument would fix all issues. You probably have a better vision about this since it depends on how operations and `basis_expansion` are converted and stored inside `OperationRepresentation` and `NoisyOperation` . 

My only message here is that, in order to have PEC working, the function `represent_operations_in_circuit*` always returns "converted" representations associated to Cirq operations. In fact this is what `execute_with_pec` actually needs for doing PEC.

Native representations would be a plus for the user, but I think has lower priority with respect to having PEC working.